File: rng.cpp
#include <iostream>
#include <vector>
#include <random>

std::vector<int> randomNumberGenerator(int min, int max, unsigned seed, int count) {
    std::mt19937 rng(seed);
    std::uniform_int_distribution<std::mt19937::result_type> dist(min, max);

    std::vector<int> numbers;
    for (int i = 0; i < count; ++i) {
        numbers.push_back(dist(rng));
    }
    return numbers;
}

int main() {
    int min = 1;
    int max = 100;
    unsigned seed = 42;
    int count = 5;

    std::vector<int> randomNumbers = randomNumberGenerator(min, max, seed, count);
    for (int number : randomNumbers) {
        std::cout << number << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: rng.java
import java.util.Random;
import java.util.ArrayList;

public class RandomNumberGenerator {
    public static ArrayList<Integer> generate(int min, int max, long seed, int count) {
        Random random = new Random(seed);
        ArrayList<Integer> numbers = new ArrayList<>();
        for (int i = 0; i < count; i++) {
            int randomNumber = random.nextInt(max - min + 1) + min;
            numbers.add(randomNumber);
        }
        return numbers;
    }

    public static void main(String[] args) {
        int min = 1;
        int max = 100;
        long seed = 42;
        int count = 5;

        ArrayList<Integer> randomNumbers = generate(min, max, seed, count);
        System.out.println(randomNumbers);
    }
}
